examples/todoapp/lib/lib_permissions/data_sources/local/permissions_local_data_source.dart

```diff
@@ -1,31 +1,23 @@
-import 'dart:convert';
-
-import '../../../app_extensions.dart';
-import '../../../base/data_sources/local/shared_preferences_instance.dart';
+import '../../../base/app/initialization/realm_instance.dart';
+import '../../models/permission_model.dart';
 
 class PermissionsLocalDataSource {
-  final SharedPreferencesInstance sharedPreferences;
-
-  PermissionsLocalDataSource({required this.sharedPreferences});
+  PermissionsLocalDataSource({required this.realmInstance});
+  final RealmInstance realmInstance;
 
-  Future<bool> savePermissions(Map<String, bool> permissions) async {
-    return await sharedPreferences.setString(
-      permissionsKey,
-      json.encode(permissions),
-    );
+  void savePermissions(PermissionModel permissions) {
+    realmInstance.realm.write(() {
+      realmInstance.realm.add<PermissionModel>(permissions);
```

Comment:
Hmm `savePermissions` sounds different than add permissions (realmInstance.realm.add<PermissionModel>(permissions)). Please make sure the permissions, which are already saved are updated based on the provided argument (PermissionModel permissions)

Reply:
Done